### PR TITLE
Create a proof of concept JAX implementation of SALT2

### PR DESCRIPTION
**Pull request by jeremykubica**
This is a minimal implementation of SALT2 using JAX. It is currently a lot slower than the sncosmo version (and can likely be optimized), but produces the correct results.

The PR includes a JAX version of sncosmo's bicubic interpolator (again this is slower than the cython version).

### CODE REVIEW

File: src/tdastro/utils/bicubic_interp.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+"""
+
+import jax.numpy as jnp
+from jax import jit, vmap
+from jax.lax import cond
+
+from tdastro.utils.io_utils import read_grid_data
+
+
+@jit
```
Comment (hombit) [inline src/tdastro/utils/bicubic_interp.py:14]: I'm not sure that we need `@jit` on this level, should we compile the code using cubic interpolation instead?

File: tests/tdastro/sources/test_salt2.py
Comment (hombit) [inline tests/tdastro/sources/test_salt2.py:1]: I'd test it vs sncosmo

**Review by hombit — COMMENTED** [review]
Looks good, but I'd ask for more testing vs implementations in other packages

File: tests/tdastro/sources/test_salt2.py
Comment (jeremykubica) [reply tests/tdastro/sources/test_salt2.py:1]: I did that during development, but I want to move away from tests that need to download sncosmo models. The download unpredictability is causing intermittent failures already. We should try to make the tests as hermetic as possible.

Comment (hombit) [reply tests/tdastro/sources/test_salt2.py:1]: Could we still run both with the test data you prepared?

File: src/tdastro/sources/salt2_jax.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        # Use the default color correction values.
+        self._colorlaw = SALT2ColorLaw.from_file(os.path.join(model_dir, cl_filename))
+
+    def _evaluate(self, phase, wavelengths, graph_state, **kwargs):
+        """Draw effect-free observations for this object.
+
+        Parameters
+        ----------
+        phase : `numpy.ndarray`
+            A length T array of rest frame timestamps.
```
Comment (mi-dai) [inline src/tdastro/sources/salt2_jax.py:98]: This is the rest frame phases (which is timestamps - time_at_peak). 
We need to add the t0 parameter like in https://github.com/lincc-frameworks/tdastro/blob/main/src/tdastro/sources/sncomso_models.py#L32
so `_evaluate()` is consistent with other physical models as `_evaluate(self, times, wavelengths ...)`

**Comment by jeremykubica** [comment]
Closing because we don't need this at the moment.

**Pull request closed without merging**